### PR TITLE
[In-Progress] Feature/itbl 2762 html in app notifcation

### PR DESCRIPTION
**Pull request by davidtruong**
Initial code for the in-app html notifications

### CODE REVIEW

File: Iterable-iOS-SDK/IterableAPI.h [hunk 1]
```diff
@@ -505,6 +505,17 @@ typedef NS_ENUM(NSInteger, PushServicePlatform) {
...
  @method
  
+ @abstract Tracks a inAppClick event
+ 
+ @param messageId       The messageId of the notification
+ @param buttonURL     The url of the button that was clicked
+ */
+- (void)trackInAppClick:(NSString *)messageId buttonURL:(NSString *)buttonURL;
+
+
```
Comment (davidtruong) [inline Iterable-iOS-SDK/IterableAPI.h]: TODO: Remove extra line

File: Iterable-iOS-SDK/IterableAPI.m [hunk 2]
```diff
@@ -846,23 +866,35 @@ - (void)spawnInAppNotification:(ITEActionBlock)callbackBlock
...
             if (message != nil) {
-                dispatch_sync(dispatch_get_main_queue(), ^{
-                    [IterableInAppManager showIterableNotification:message trackParams:notification callbackBlock:(ITEActionBlock)callbackBlock];
-                });
+                NSString *html = [message objectForKey:@"html"];
+                if ([html length] != 0) {
+                    if ([html containsString:@"href"]) {
+
+                        NSDictionary *inAppDisplaySettings = [message valueForKey:@"inAppDisplaySettings"];
+                        double backgroundAlpha = .5f; //[[inAppDisplaySettings valueForKey:@"backgroundAlpha"] doubleValue];
```
Comment (davidtruong) [inline Iterable-iOS-SDK/IterableAPI.m]: Get value from the payload instead of hardcoding it.

File: Iterable-iOS-SDK/IterableConstants.h [hunk 3]
```diff
@@ -70,6 +70,8 @@ extern NSString *const ITBL_DEVICE_USER_INTERFACE;
 
 
 //In-App Constants
+#define ITERABLE_IN_APP_CLICK_URL @"urlClick"
```
Comment (davidtruong) [inline Iterable-iOS-SDK/IterableConstants.h:76]: New payload key used for tracking url clicks on in-app notifications

**Comment by davidtruong** [comment]
I'll start the review now but I still need to do final testing whenever the backend is up on staging.

File: Iterable-iOS-SDK/IterableAPI.m [hunk 4]
```diff
@@ -846,23 +866,35 @@ - (void)spawnInAppNotification:(ITEActionBlock)callbackBlock
...
+                } else {
+                    LogWarning(@"Empty html found in inApp content %@", message);
+                }
             }
         } else {
             LogDebug(@"No notifications found for inApp payload %@", payload);
         }
     };
-
+
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableAPI.m:895]: whitespace

**Review by dontgitit — APPROVED** [review]
skimmed through, nothing stood out

**Pull request merged**